File: src/renderer/Models/Section.ts
export default class Section {
  descriptionLines: string[];

  constructor(descriptionLines: string[] = []) {
    this.descriptionLines = descriptionLines;
  }

  parseDescriptionLine(line: string): void {
    const data = line.match(/\*\s{2}(.+)\r?\n?/);
    if (data !== null) this.descriptionLines.push(data[1]);
  }

  format(): string {
    let text = '';
    text = text.concat('****************************************');
    text = text.concat('\n');
    this.descriptionLines.forEach((element) => {
      text = text.concat(`*  ${element}\n`);
    });
    text = text.concat('\n****************************************');
    return text;
  }
}

File: src/app_table_manager/models/section.cpp
#include "models/section.h"

#include <regex>

Section::Section(vector<string> descriptionLines) {
  description = vector<string>(descriptionLines);
}

void Section::ParseDescriptionLine(string line) {
  std::regex e("\\*\\s{2}(.+)\\r?\\n?");
  std::smatch matchSet;

  std::regex_match(line, matchSet, e);

  if (matchSet.size() == 2)
    description.push_back(matchSet[1]);
}

string Section::Format(int id) {
  string text = "";

  text += "****************************************";
  text += '\n';
  for (string line : description) {
    text += "*  ";
    text += line;
    text += '\n';
  }
  text += '\n';
  text += "****************************************";

  return text;
}

vector<string> Section::GetDescription() {
  return description;
}
